sirepo/package_data/static/js/sirepo.js

```diff
@@ -1907,13 +1963,35 @@ SIREPO.app.controller('SimulationsController', function (appState, panelState, r
     }
 
     function setActiveFolder(item) {
+        var prevPath = self.pathName(self.activeFolder || item);
         self.activeFolder = item;
         self.activeFolderPath = [];
         while (item) {
             self.activeFolderPath.unshift(item);
             item = item.parent;
         }
         appState.setActiveFolderPath(self.pathName(self.activeFolder));
+        appState.setActiveFolder(self.activeFolder);
+        if( prevPath !== self.pathName(self.activeFolder) ) {
+            var compoundPath = pathToCompoundPath(self.pathName(self.activeFolder));
```

Comment:
pathToCompoundPath can return '/' so it seems if you want to check if it is root folder first before calling compoundPath.